zuul-core/src/main/java/com/netflix/zuul/util/HTTPRequestUtils.java

```diff
@@ -133,15 +133,31 @@ public String getFormValue(String sHeaderName) {
         if(headerNames != null) {
             while (headerNames.hasMoreElements()) {
                 String name = headerNames.nextElement();
-                String value = request.getHeader(name);
-
-                if(name != null && !name.isEmpty() && value != null) {
-                    List<String> valueList = new ArrayList<String>();
-                    if(headers.containsKey(name)) {
-                        headers.get(name).add(value);
+                if (name != null && !name.isEmpty()){
+                    Enumeration values = request.getHeaders(name);
+                    if (values == null){
+                        continue;
+                    }
+                    if (headers.get(name) != null){
+                        while (values.hasMoreElements()){
+                            Object value = values.nextElement();
+                            if (value == null){
+                                continue;
```

Comment:
Instead of continuing, you can do the inverse of the if statement and add it then. It'll make the code more readable. You can replace all of the `continue`s that way.

Reply:
yes,you are right